### DISCUSSION TITLE
Add Device_Class "Window" for Sensors

### DISCUSSION THREAD

**Discussion by ChristophCaina** [home-assistant/architecture#842] [General] [answered]
I do have some Sensors that provide the state of a Window-Handle.
Previously, I've used the core-integration "Homematic IP" for these Handles - and they have reported only "open" / "close" as binary-sensor.

I've moved the whole Installation from Homematic IP (Cloud-based) to a local CCU - by using the Custom Integration "Homematic (IP) local"...
With this integration, the sensor does provide three states:

- Closed
- Open
- Tilted

I know, that there are also "Smart Window Handles" from Companies like enOcean, for which I think, they should report their status similar.

It would be great, when a Device_Class "Window" or "WindowHandle" could be implemented for those Devices, in Order to extend the capabilities with Coloring and Icon Change, etc.

Here's the code from the Core EnOcean Integration...

```
class EnOceanWindowHandle(EnOceanSensor):
    """Representation of an EnOcean window handle device.
    EEPs (EnOcean Equipment Profiles):
    - F6-10-00 (Mechanical handle / Hoppe AG)
    """

    def value_changed(self, packet):
        """Update the internal state of the sensor."""
        action = (packet.data[1] & 0x70) >> 4

        if action == 0x07:
            self._attr_native_value = STATE_CLOSED
        if action in (0x04, 0x06):
            self._attr_native_value = STATE_OPEN
        if action == 0x05:
            self._attr_native_value = "tilt"
```

**Comment by ReneNulschDE** [comment]
Whenever this gets reviewed, I would propose the add few more state to support windows of cars too:
"window completely closed" = STATE_CLOSED

missing:
"window completely opened"
"window in intermediate position"
"window airing position" (
"window intermediate airing position"
"window currently running"

**Comment by frenck** [comment, marked as answer]
I don't think we should take sensor device classes for this, as it is prone to a legion of different states across multiple devices. For that reason, almost all sensor device classes are for numeric state.

For non-numeric state with an fixed list of possibilities, we recently added the `enum` device class. For those sensors, the list of known textual state can be provided upfront (and thus used in automation, scripts and translation).

../Frenck
